### DISCUSSION TITLE
Why does linting need a write to update?

### DISCUSSION THREAD

**Discussion by txtyash** [helix-editor/helix#1208] [Q&A] [answered]
While coding the errors and warnings are updated as soon as I open the file but once I start editing the code then I have to write to the file to see updated linting like new errors and clear out the linting for those that are fixed.

Sometimes it does auto-update tho otherwise it needs a write to file.

**Comment by archseer** [comment, marked as answer]
That's up to the language server. In the case of rust some checks are computationally expensive and only done on save.
